Source/WTF/wtf/glib/FileSystemGlib.cpp

```diff
@@ -195,10 +195,18 @@ PlatformFileHandle openFile(const String& path, FileOpenMode mode, FileAccessPer
 
     if (mode == FileOpenMode::Read)
         ioStream = adoptGRef(g_file_open_readwrite(file.get(), nullptr, nullptr));
-    else if (mode == FileOpenMode::Write || mode == FileOpenMode::ReadWrite) {
-        if (g_file_test(filename.data(), static_cast<GFileTest>(G_FILE_TEST_EXISTS | G_FILE_TEST_IS_REGULAR)))
-            ioStream = adoptGRef(g_file_open_readwrite(file.get(), nullptr, nullptr));
-        else
+    else {
+        ASSERT(mode == FileOpenMode::Truncate || mode == FileOpenMode::ReadWrite);
+        bool fileExists = g_file_test(filename.data(), static_cast<GFileTest>(G_FILE_TEST_EXISTS | G_FILE_TEST_IS_REGULAR));
+        if (fileExists) {
+            if (mode == FileOpenMode::ReadWrite)
+                ioStream = adoptGRef(g_file_open_readwrite(file.get(), nullptr, nullptr));
+            else {
+                deleteFile(path);
+                fileExists = false;
+            } 
+        }
+        if (!fileExists)
```

Comment:
I think `g_file_replace_readwrite()` can be used here for truncation mode, and the whole logic of this function could probably be reworked a bit to accommodate it. Somebody else can swoop in and do it after this change.

Reply:
I took a look and I think the code should just be:

```
    if (mode == FileOpenMode::Read || FileOpenMode::ReadWrite)
        ioStream = adoptGRef(g_file_open_readwrite(file.get(), nullptr, nullptr));
    else {
        ASSERT(mode == FileOpenMode::Truncate);
        ioStream = adoptGRef(g_file_replace_readwrite(file.get(), nullptr, FALSE, permissionFlag, nullptr, nullptr));
    }
```

I'm building now to make sure that compiles.

(P.S. I also noticed that it's weird to convert FileOpenMode::Read to a read/write, but the return value has to be a GIOStream, which is inherently writable, so it's not actually possible to respect the FileOpenMode. To do actual read-only or write-only opens, we'd need to return GInputStream or GOutputStream instead. Then PlatformIOHandle would have to be a GObject rather than a GIOStream. Probably not worth it.)

Reply:
Fixed a build error. It doesn't immediately explode at runtime either, so decent chance it's even correct....

Reply:
Thanks for the review; I updated Source/WTF/wtf/glib/FileSystemGlib.cpp with the proposed approach.